### PR TITLE
Form: Add custom template for array fields

### LINKED ISSUE #473: Improve display of arrays in Form component
**Describe the solution you'd like**
At the moment the buttons used to manipulate array elements in a form don't have any text or icons, making it very hard to use. We need to improve this to make it more usable.

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Fixes #473

Change-type: minor
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

### CODE REVIEW

File: src/unstable/components/Form/ArrayFieldTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	const id = `${idSchema.$id}__description`;
+	return <DescriptionField id={id} description={description} />;
+};
+
+const ActionButton = styled(Button)`
+	color: ${({ theme }) => theme.colors.text.light};
+`;
+
+export default (props: ArrayFieldTemplateProps) => {
+	console.log(props);
```
Comment (thgreasi) [inline src/unstable/components/Form/ArrayFieldTemplate.tsx]: `console.log` :police_car:

File: src/stories/beta/Form.js [hunk 2]
```diff
@@ -36,16 +36,24 @@ const basicPokedexSchema = {
...
       type: 'boolean'
     },
     first_seen: {
       title: 'First seen',
+      description: 'The first time you saw this pokèmon',
       type: 'string',
       format: 'date-time'
+    },
+    Tags: {
+      description: 'Add useful tags to your pokèmon',
```
Comment (thgreasi) [inline src/stories/beta/Form.js]: NIT: how about using a title here for consistency?

File: src/stories/beta/Form.js [hunk 3]
```diff
@@ -36,16 +36,24 @@ const basicPokedexSchema = {
...
     caught: {
       type: 'boolean'
     },
     first_seen: {
       title: 'First seen',
+      description: 'The first time you saw this pokèmon',
       type: 'string',
       format: 'date-time'
+    },
+    Tags: {
```
Comment (thgreasi) [inline src/stories/beta/Form.js]: The `Tags` label is rendered differently when compared with the rest of the titles (eg `Tags` is bold while `FIRST SEEN` isn't and shows up in uppercase).
How about re-using the `FieldTemplate.Label`?

**Comment by LucianBuzzo** [comment]
@thgreasi I've addressed all your feedback comments

**Comment by thgreasi** [comment]
As an added bonus, would it make sense to auto-focus the newly added array fields?
As a side-note, on FF the new field get's the "required" red border
image

**Comment by LucianBuzzo** [comment]
@thgreasi I've added new issues to address your feedback

**Pull request merged**